src/interlocking_component/route_controller.py

```diff
@@ -106,16 +109,53 @@ class RouteController(IRouteController):
     It makes sure, that the Interlocking sets fahrstrassen along those routes.
     """
 
-    def is_new_fahrstrasse_needed(self, train: "Train", track_segment: "Track"):
+    interlocking: interlockinginterface = None
+    router: Router = None
+
+    def is_new_fahrstrasse_is_needed(self, train: Train, track: Track) -> bool:
         """This method should be called when a train enters a new track_segment.
         It then checks if the train is near the end of his fahrstrasse and updates it, if necessary.
 
         :param train: the train that may need a new fahrstasse
         :type Train: Train
-        :param track_segment: the track_segment it just entered
+        :param track_segment: the track it just entered
         :type Track: Track
         """
-        raise NotImplementedError()
+        route = None
+        for route_candidate in self.interlocking.active_routes:
+            interlocking_track_candidat = route_candidate.contains_segment(
+                track.identifier
+            )
+            # This assumes, that track.identifier does not contain '-re',
+            # as the interlocking does not know of reverse directions.
+            if interlocking_track_candidat is not None:
+                route = route_candidate
+        if route is None or route.get_last_segment_of_route != track.identifier:
+            return False
+
+        new_route = self.router.get_route(track, train.timetable[0].track)
+        for end_node_candidat in new_route:
+            for interlocking_route in self.interlocking.routes:
+                if (
+                    interlocking_route.start_signal.name == new_route[0]
```

Comment:
I love your comments 💚 That makes is easy to understand the code!
Could you please add a small comment to this statement too? 🙃